packages/opentelemetry-core/test/context/B3Propagator.test.ts

```diff
@@ -83,6 +89,59 @@ describe('B3Propagator', () => {
       );
       assert.deepStrictEqual(carrier[X_B3_SPAN_ID], '6e0c63257de34c92');
       assert.deepStrictEqual(carrier[X_B3_SAMPLED], '0');
+      assert.deepStrictEqual(carrier[X_B3_FLAGS], undefined);
+      assert.deepStrictEqual(carrier[X_B3_PARENT_SPAN_ID], undefined);
+    });
+
+    it('should set set b3 traceId, spanId, and flags headers', () => {
```

Comment:
can you split this into more unit tests so when unit test fails it will be straightforward what has happened? Ideally with "when -> then" approach
```
desribe('when ...', ()=>{
  beforeEach(()=> {
    // prepare data
  });
  it('should set spanId', ()=>{});
  it('should set flags', ()=>{});
  it('should set sampled', ()=>{});
  it('should set parentSpanId', ()=>{});
})
```

Reply:
I can do it for the new tests I added in extract, but this is already in a describe block for injects that only has 5 tests, and it doesn't seem that beneficial to do it here.